cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/runtimejob/DefaultRuntimeJob.java

```diff
@@ -103,6 +103,7 @@ public void run(RuntimeJobEnvironment runtimeJobEnv) throws Exception {
     // Create cConf with provided properties. These properties can be used to set configs for twill runner such as
     // connection string for discovery.
     CConfiguration cConf = createCConf(runtimeJobEnv);
+    System.setProperty(Constants.SPARK_COMPAT_ENV, cConf.get(Constants.AppFabric.SPARK_COMPAT));
```

Comment:
Actually spark version should be passed to DataprocJobMain from provisioner. This is because provisioner supports multiple spark compat based on image version
https://github.com/cdapio/cdap/blob/develop/cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java#L179
Let me know your thoughts

Reply:
Yes, that's what I meant by returned from the job env properties. It is the DataprocProvisioner who creates the RuntimeJobManager, which in turn launch the job.

Reply:
sounds good. 👍